sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/StateStoreRDDSuite.scala

```diff
@@ -227,6 +228,32 @@ class StateStoreRDDSuite extends SparkFunSuite with BeforeAndAfter {
     }
   }
 
+  test("SPARK-51823: unload on commit") {
```

Comment:
Should we add an integration test under RocksDBStateStoreIntegrationSuite with the config enabled ?

Reply:
Added a basic integration test, let me know if there's anything you want to add to it